icefoganalytics/travel-authorization
# Wire Pre-Approved Travel to Travel Details

Fixes https://github.com/icefoganalytics/travel-authorization/issues/169

Relates to:

- https://github.com/icefoganalytics/travel-authorization/pull/179

# Context

In the Approval Region of the travel details tab (https://travel-auth-dev.ynet.gov.yk.ca/my-travel-requests/78/details) can you wire the Pre-approved Travel request drop down to show Approved preapproved travel for a user based on department.

![image](https://github.com/icefoganalytics/travel-authorization/assets/89539008/264d4513-0afc-477f-a7b3-2655e4afe24d)

# Implementation

Rename `preapprovedTravelers` to `travel_authorization_pre_approval_profiles` to better reflects the mix of user and user-group information.

# Screenshots

Edit view of multiple options for pre-approval profile selection
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/130d3008-591d-4cfb-aa4e-67eb2956f49b)

Read-only view of pre-approval profile reference
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/b29eaccb-95dc-4ee4-86d5-63cffa43563b)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Select "PreApproved" from the top nav drop down.
5. Click the "Add new travel" button to begin adding a new pre-approval.
6. Set a start date in the future, and add a few travelers.
7. After creating the pre-approval, select it via the checkbox, then click the "Submit selected travel" button.
8. Click "submit" in the modal.
9. Now go to the "Submissions" tab, and click "Approve" on the thing you just submitted.
10. Type in an approver and approval date.
11. Upload some random PDF to simulate a signed document.
12. Change the status to "approve"
13. Click the "save" button.
14. Next go to the "My Travel Requests" via the top drop down nav.
15. Create a new travel authorization.
16. Fill in all the details.
17. Check that you have some entries in the "pre-approved travel for (if applicable)" selector. Pick an option.
18. Submit the travel authorization to yourself.
19. After the submission, check that the "pre-approved travel for (if applicable)" field is filled in correctly.
